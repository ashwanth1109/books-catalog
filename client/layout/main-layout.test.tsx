import { act, render } from "@testing-library/react";
import MainLayout from "./main.layout";

describe("main.layout.tsx", () => {
  it("should assert that main layout snapshot is correct", async () => {
    await act(async () => {
      const { container } = render(<MainLayout />);

      await new Promise((resolve) => setTimeout(resolve, 0));

      expect(container).toMatchSnapshot();
    });
  });
});
