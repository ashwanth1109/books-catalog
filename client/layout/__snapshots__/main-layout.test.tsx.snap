// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main.layout.tsx should assert that main layout snapshot is correct 1`] = `
<div>
  <section
    class="ant-layout ant-layout-has-sider min-h-screen"
  >
    <aside
      class="ant-layout-sider ant-layout-sider-dark ant-layout-sider-has-trigger min-h-screen"
      style="flex: 0 0 200px; max-width: 200px; min-width: 200px; width: 200px;"
    >
      <div
        class="ant-layout-sider-children"
      >
        <div
          class="w-full h-24"
        />
        <ul
          class="ant-menu ant-menu-root ant-menu-inline ant-menu-dark"
          data-menu-list="true"
          role="menu"
          tabindex="0"
        >
          <li
            class="ant-menu-item ant-menu-item-selected"
            data-menu-id="rc-menu-uuid-test-1"
            role="menuitem"
            style="padding-left: 24px;"
            tabindex="-1"
          >
            <span
              aria-label="file"
              class="anticon anticon-file ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="file"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M854.6 288.6L639.4 73.4c-6-6-14.1-9.4-22.6-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.7-9.4-22.7zM790.2 326H602V137.8L790.2 326zm1.8 562H232V136h302v216a42 42 0 0042 42h216v494z"
                />
              </svg>
            </span>
            <span
              class="ant-menu-title-content"
            >
              Books
            </span>
          </li>
        </ul>
        <div
          aria-hidden="true"
          style="display: none;"
        />
      </div>
      <div
        class="ant-layout-sider-trigger"
        style="width: 200px;"
      >
        <span
          aria-label="left"
          class="anticon anticon-left"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="left"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
            />
          </svg>
        </span>
      </div>
    </aside>
    <section
      class="ant-layout"
    >
      <div
        class="w-full h-16 bg-white"
      />
      <main
        class="ant-layout-content mx-8"
      />
    </section>
  </section>
</div>
`;
